packages/klighd-core/src/search-bar/searchbar-actions.ts

```diff
@@ -0,0 +1,680 @@
+        return {
+            kind: KIND,
+            panel,
+        }
+    }
+
+    export function isThisAction(action: Action): action is RetrieveTagsActions {
+        return action.kind === KIND
+    }
+}
+
+export interface SearchAction extends Action {
+    kind: typeof SearchAction.KIND
+    id: string
+    panel: SearchBarPanel
+    textInput: string
+    tagInput: string
+}
+
+// eslint-disable-next-line no-redeclare
+export namespace SearchAction {
+    export const KIND = 'handleSearch'
+
+    export function create(panel: SearchBarPanel, id: string, textInput: string, tagInput: string): SearchAction {
+        return {
+            kind: KIND,
+            id,
+            textInput,
+            tagInput,
+            panel,
+        }
+    }
+
+    export function isThisAction(action: Action): action is SearchAction {
+        return action.kind === KIND
+    }
+}
+
+interface HighlightBounds {
+    x: number
+    y: number
+    width: number
+    height: number
+}
+
+@injectable()
+export class HandleSearchAction implements IActionHandler {
+    private static currentModel?: SModelElement
+
+    private OPACITY_INCREMENT: number = 2
+
+    private HIGHLIGHT_MATCH: number = 2
+
+    private HIGHLIGHT_MAIN_MATCH: number = 1
+
+    // TODO: ktexts can't have a border, so instead of setting highlight directly on the ktext, a rectangle with the correct
+    //       size should be added behind it instead (this does pose an additional issue with the foreground then not being
+    //       applied to the text itself, so it's a choice, support border or support foreground highlight)
+
+    @inject(TYPES.IActionDispatcher) private actionDispatcher: IActionDispatcher
+
+    initialize(registry: ActionHandlerRegistry): void {
+        registry.register(SetModelAction.KIND, this)
+        registry.register(UpdateModelAction.KIND, this)
+        registry.register(SearchAction.KIND, this)
+        registry.register(ClearHighlightsAction.KIND, this)
+        registry.register(UpdateHighlightsAction.KIND, this)
+        registry.register(RetrieveTagsAction.KIND, this)
+    }
+
+    handle(action: Action): void {
+        /* Intercept model during SetModelAction / UpdateModelAction */
+        if (action.kind === SetModelAction.KIND || action.kind === UpdateModelAction.KIND) {
+            const model = (action as SetModelAction).newRoot as SModelElement
+            HandleSearchAction.currentModel = model
+            return
+        }
+
+        if (!HandleSearchAction.currentModel) return
+
+        const modelId = HandleSearchAction.currentModel?.id
+
+        /* Handle ClearHighlightsActions */
+        if (ClearHighlightsAction.isThisAction(action)) {
+            removeHighlights(HandleSearchAction.currentModel)
+            // make changes visible
+            if (modelId && this.actionDispatcher) {
+                this.actionDispatcher.dispatch(CenterAction.create([modelId]))
+            }
+        }
```

Comment:
We always center the view if we clear the search highlights? I did not see this happen, but wonder why this should happen in the first place.